x-pack/plugins/reporting/server/routes/generate/csv_searchsource_immediate.ts

```diff
@@ -73,7 +73,12 @@ export function registerGenerateCsvFromSavedObjectImmediate(
         const counters = getCounters(req.route.method, path, reporting.getUsageCounter());
 
         const logger = parentLogger.get(CSV_SEARCHSOURCE_IMMEDIATE_TYPE);
-        const runTaskFn = runTaskFnFactory(reporting, logger);
+        const csvSearchSourceImmediateExport = reporting
+          .getExportTypesRegistry()
+          .get(
+            (e) => e.jobType === 'csv_searchsource_immediate'
+          ) as unknown as CsvSearchSourceImmediateExportType;
```

Comment:
Going along with my previous comment, rather than collecting the csvSearchSourceImmediateExport object from the export types registry, we should forgot the idea that this object fits into our registry, and in this code, have something like:
```
const csvSearchSourceImmediateExport = reporting.getSearchSourceImmediateExport()
```

This makes better isolation of this troublesome export type, so the unique shape of it doesn't impact other "normal" code.

Reply:
Note for future me - setup() and start() for csvSearchSource immediate may need to be set up in genereate/csv_searchsource_immediate.ts